apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -205,7 +213,7 @@ where
                         tx_result["code"] = ErrorCodes::InvalidTx.into();
                     }
                     tx_result["gas_used"] = result.gas_used.to_string();
-                    tx_result["info"] = result.to_string();
+                    tx_result["info"].push_str(&result.to_string());
```

Comment:
I'm not sure if we should be appending strings like this as it's gonna make the tx results harder to parse. Ideally it should all go via `TxResult`

Reply:
I guess that would include all the tx error conditions (we have this issue to refactor it https://github.com/anoma/anoma/issues/455)

Reply:
So the error code is enough to know that it's not decryptable. So I can remove the extra info line. I'm not entirely happy though because all other error codes mean that the proposed block should be voted against by validators. This is an accepting code path (so not an error from the protocol pov), but it is an error from the user pov

Reply:
Yeah, it would be good to preserve the information. Could we maybe add it as an optional field to the `TxResult` so that it can still be parsed from the "info" field?

Reply:
I've added a log field